src/lib.rs

```diff
@@ -140,6 +162,20 @@ macro_rules! define_matrix {
             }
         }
 
+        impl From<$mint_type> for $name {
+            #[inline(always)]
+            fn from(other: $mint_type) -> Self {
+                Self {
+                    inner: [
+                        $(
```

Comment:
Instead of using an array of column arrays, if you explicitly convert into an array with length M * N, you should be able to copy the conversion code directly from above. Also no extra parameters.